foreshadow/transformers/base.py

```diff
@@ -399,7 +406,18 @@ def fit(self, X, y=None, **kwargs):
         else:
             self._verify_transformer(X, y, refit=True)
         inject = _inject_df(self.transformer, kwargs.pop("full_df", None))
-        return self.transformer.fit(X, y, **{**kwargs, **inject})
+
+        # Check if using a sklearn transformer with only y vars
+        if "X" not in inspect.getfullargspec(self.transformer.fit).args:
+            return self.transformer.fit(X, **{**kwargs, **inject})
```

Comment:
Shouldn't the only argument passed into fit() be 'y' instead of 'X'?

Reply:
Here X is actually the "y" variable that is passed into the function.

Reply:
https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html#sklearn.preprocessing.LabelEncoder.fit